Module-09 Template and Visualforce Component/classes/QueryComponentController.cls

```diff
@@ -0,0 +1,161 @@
+                listOfDisplayingObjects = getDisplayFields(queryToBeExecuted);
+                if(listOfDisplayingObjects != null)
+                {
+                    setCon = new ApexPages.StandardSetController(listOfDisplayingObjects);
+                    setCon.setPageSize(10);
+                }
+            }
+            
+            return setCon;
+        }
+        set;
+    }
+    
+    public QueryComponentController()
+    {
+        this.queryToBeExecuted = '';
+        this.sortBy = '';
+    }
+   
+    public String getQueryToBeExecuted()
+    {
+        return this.queryToBeExecuted;
+    }
+    
+    public void setQueryToBeExecuted(String query)
+    {
+        if(!this.queryToBeExecuted.equalsIgnoreCase(query))
+        {
+            this.queryToBeExecuted = query;
+            this.setCon = null;
+        }
+    }
+    
+    public void sortByMethod()
+    {
+        this.setCon = null;
+        this.isSortingLinkClicked = true;
+        this.sortAscending = !this.sortAscending; 
+    }
+    
+    public void setSortBy(String fieldName)
+    {
+        this.sortBy = fieldName;
+    }
+
+    public String getSortBy()
+    {
+        return this.sortBy;
+    }
+    
+	public List<sObject> getRecordsList()
+    {
+        if(setCon != null)
+        {
+            return setCon.getRecords();
+        }
+        
+        else
+        {
+            return null;
+        }
+    }
+    
+    private String createQuery(String query)
+    {
+        String orderedQuery = query;        
+        if(isSortingLinkClicked && String.isNotBlank(this.getSortBy()))
+        {
+            if(sortAscending)
+            {
+                if(query.containsIgnoreCase('order by'))
+                {
+                    orderedQuery = orderedQuery.substring(0, query.indexOfIgnoreCase('order by') - 1);
+                }
+                orderedQuery = orderedQuery + ' ORDER BY ' + this.getSortBy();                
+            }
+            
+            else
+            {
+                if(query.containsIgnoreCase('order by'))
+                {
+                    orderedQuery = orderedQuery.substring(0, query.indexOfIgnoreCase('order by') - 1);
+                }
+                orderedQuery = orderedQuery + ' ORDER BY ' + this.getSortBy() + ' DESC';
+            }
+        }
+        
+        return orderedQuery;
+    }
+    
+    private List<sObject> executeQuery(String query)
+    {
+        String orderedQuery = createQuery(query);
+        List<sObject> sObjectList = new List<sObject>();
+        try
+        {
+            sObjectList = Database.query(orderedQuery);
+            
+        }
+        catch(Exception ex)
+        {
+            ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.ERROR, ex.getMessage())); 
+            
+            return null;
+        }
+        
+        return sObjectList;
+    }
+    
+    public List<sObject> getDisplayFields(String query)
```

Comment:
method name suggests that it will provide fields but it is providing sObject. 
confused method